### PR TITLE
[proto] Added functional `perspective_bounding_box/segmentation_mask` ops

### PR DESCRIPTION
**Pull request by vfdev-5**
Related to #5514 and #5782

Description:
- Added functional `perspective_bounding_box/segmentation_mask` ops
- Added tests

### Results on synthetic images/bboxes/segm mask:

<details>
<summary>
Code
</summary>

```python
import numpy as np

import torch
import torchvision
from torchvision.prototype import features
from torchvision.prototype.transforms.functional import perspective_bounding_box, perspective_image_tensor, perspective_segmentation_mask

size = (64, 76)
# xyxy format
in_boxes = [
    [10, 15, 25, 35],
    [50, 5, 70, 22],
    [45, 46, 56, 62],
]
labels = [1, 2, 3]

im1 = 255 * np.ones(size + (3, ), dtype=np.uint8)
mask = np.zeros(size, dtype=np.int64)
for in_box, label in zip(in_boxes, labels):
    im1[in_box[1]:in_box[3], in_box[0]:in_box[2], :] = (127, 127, 127)
    mask[in_box[1]:in_box[3], in_box[0]:in_box[2]] = label
    
in_im = torch.tensor(im1).permute(2, 0, 1).view(1, 3, *size)

in_boxes = features.BoundingBox(
    in_boxes, format=features.BoundingBoxFormat.XYXY, image_size=size
)
in_mask = features.SegmentationMask(torch.tensor(mask)).view(1, *size)
    

coeffs = [3.7801878452301025,
 2.09562087059021,
 -59.95088195800781,
 0.9144350290298462,
 2.9016025066375732,
 -37.96672821044922,
 0.0310051292181015,
 0.01576310582458973]

out_boxes = perspective_bounding_box(
    in_boxes, 
    in_boxes.format,
    coeffs
)

out_mask = perspective_segmentation_mask(
    in_mask, 
    coeffs
)

out_im = perspective_image_tensor(in_im, coeffs, interpolation=torchvision.transforms.InterpolationMode.NEAREST)
```

</details>

image

### CODE REVIEW

**Review by datumbox — COMMENTED** [review]
@vfdev-5 Thanks a lot for the PR, just a couple of minor comments. Let me know your thoughts.

@pmeier any thoughts on your end?

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 1]
```diff
@@ -458,6 +458,81 @@ def perspective_image_pil(
...
+        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY
+    ).view(-1, 4)
+
+    dtype = bounding_box.dtype if torch.is_floating_point(bounding_box) else torch.float32
+    device = bounding_box.device
+
+    # perspective_coeffs are computed as endpoint -> start point
+    # We have to invert perspective_coeffs for bboxes
+    denom = perspective_coeffs[0] * perspective_coeffs[4] - perspective_coeffs[1] * perspective_coeffs[3]
+    # TODO: assert that is not zero
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply torchvision/prototype/transforms/functional/_geometry.py:481]: perspective_coeffs are on CPU

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 2]
```diff
@@ -458,6 +458,81 @@ def perspective_image_pil(
...
+
+    inv_coeffs = [
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply torchvision/prototype/transforms/functional/_geometry.py:483]: OK, I'll add a comment.
It is just an inversion of perspective transformation:
```
x_out = (coeffs[0] * x + coeffs[1] * y + coeffs[2]) / (coeffs[6] * x + coeffs[7] * y + 1)
y_out = (coeffs[3] * x + coeffs[4] * y + coeffs[5]) / (coeffs[6] * x + coeffs[7] * y + 1)
```
into
```
x = (inv_coeffs[0] * x_out + inv_coeffs[1] * y_out + inv_coeffs[2]) / (inv_coeffs[6] * x_out + inv_coeffs[7] * y_out + 1)
y = (inv_coeffs[3] * x_out + inv_coeffs[4] * y_out + inv_coeffs[5]) / (inv_coeffs[6] * x_out + inv_coeffs[7] * y_out + 1)
```

**Comment by vfdev-5** [comment]
`test/test_prototype_transforms_functional.py::test_correctness_perspective_bounding_box` can be flaky on windows:
- https://github.com/pytorch/vision/runs/6254487636?check_suite_focus=true
- https://github.com/pytorch/vision/runs/6254835280?check_suite_focus=true (just restarted)
- https://github.com/pytorch/vision/runs/6255044138?check_suite_focus=true

**Review by datumbox — APPROVED** [review]
LGTM, thanks @vfdev-5.

@pmeier Any thoughts on the testing side?
